src/channel.rs

```diff
@@ -52,3 +64,169 @@ impl fmt::Debug for ChannelData {
             .finish()
     }
 }
+
+#[derive(Debug, Default)]
+pub(crate) struct ChannelHandler {
+    allocations: Vec<ChannelAllocation>,
+}
+
+#[derive(Debug)]
+struct ChannelAllocation {
+    id: ChannelId,
+
+    /// Stream id, when it is known. This might be delayed awaiting sctp initialization to
+    /// know if we are client or server.
+    sctp_stream_id: Option<u16>,
+
+    /// Holds the config until it is used in handle_timeout.
+    config: Option<ChannelConfig>,
+}
+
+impl ChannelHandler {
+    pub fn new_channel(&mut self, config: &ChannelConfig) -> ChannelId {
+        let id = ChannelId(self.allocations.len());
+
+        // For out-of-band negotiated, the id is already set.
+        let sctp_stream_id = config.negotiated;
+        if let Some(sctp_stream_id) = sctp_stream_id {
+            let exists = self
+                .allocations
+                .iter()
+                .any(|a| a.sctp_stream_id == Some(sctp_stream_id));
+            assert!(exists, "sctp_stream_id ({}) exists already", sctp_stream_id);
```

Comment:
should this be !exists?

Reply:
No? I mean to check that one doesn't set a channel that already has been set before automatically or manually.

Reply:
You're right! Fixed.